src/Main.elm

```diff
@@ -1,10 +1,829 @@
+
+           And additionally, some things are so much more easily expressed with
+           elm-ui.
+           For example: an entire (working) empty page example that displays
+           centered text:
+           ```elm
+           view = E.layout
+               [
+               ]
+               <| E.el
+                   [ E.centerX
+                   , E.centerY
+                   ]
+                   <| E.text "hello"
+           ```
+           Which will center the text.
+           With css, even a simple task like this is so unintuitive that there's a
+           very popular thread on stack overflow on how to do it, with answers of
+           varying complexity that I would argue are way too cryptic for the
+           simplicity of the problem.
+           [The thread](https://stackoverflow.com/questions/114543/how-to-horizontally-center-a-div)
+
+   Possible counter arguments:
+        1. What if the user has javascript disabled? Then the web page won't even
+           display!
+
+           Answer:
+           My rebuttal to this is that since this is an aplication that requires
+           javascript to even work, this is not an issue. No point in seeing
+           the app if you can't use it.
+           Also, even if we used elm/html, and we had css separately, if javascript
+           doesn't run, no html nodes will even be created, so again, nothing will
+           be displayed. So both approaches suffer from the same problem.
+
+   It's difficult to convey through these arguments how much more joyful I am
+   styling my app in elm-ui versus css, but it's such a big deal for me that this
+   library was one of the main reasons I picked up elm.
+
+   Also relevant, Matt Griffith's talk about the elm-ui library:
+       https://www.youtube.com/watch?v=Ie-gqwSHQr0
+-}
+
+
+view : Model -> Html.Html Msg
+view model =
+    let
+        content =
+            case model of
+                Loading loadingModel ->
+                    viewLoading loadingModel
+
+                Loaded loadedModel ->
+                    viewDashboard loadedModel
+
+                JsonError err ->
+                    viewJsonError err
+    in
+    E.layout
+        -- this is the root of the page
+        [ EFont.family
+            -- we set a font family that applies globally
+            [ EFont.typeface Palette.font0
+            ]
+        , EFont.size Palette.fontSize0 -- and a size
+        , EBackground.color Palette.white -- and the background color of the whole page
+        ]
+        -- we split the page into a column of the content and
+        -- the color palette at the bottom
+        <| E.column
+            [ E.width E.fill
+            , E.height E.fill
+            ]
+            [ E.el [ E.paddingXY 0 80, E.centerX, E.height E.fill ] content
+            , E.row
+                [ E.width E.fill
+                , E.alignBottom
+                ]
+                <| List.map viewColor colors
```

Comment:
So if I'm right, this piece, with some code in Palette and with `colors` list is producing bunch of elements aligned by `flex` to produce basically this CSS baground, right?
```CSSS
background: linear-gradient(90deg, rgba(10,30,60,1) 0%, rgba(10,30,60,1) 16.6%, rgba(20,46,80,1) 16.6%, rgba(38,74,120,1) 33.2%, rgba(20,46,80,1) 33.2%, rgba(38,74,120,1) 49.8%, rgba(90,150,225,1) 49.8%, rgba(90,150,225,1) 66.4%, rgba(130,180,235,1) 66.4%, rgba(130,180,235,1) 83%, rgba(192,228,255,1) 83%, rgba(192,228,255,1) 100%);
```
Not sure which approach is more obvious from code 🤷

Reply:
I would prefer the CSS way here as well. Or having this in separate function with some name like "fancy gradient background" would work as well here for me :)